core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -374,13 +375,17 @@ static HttpRequestInterceptor loadInterceptorDynamically(
     return instance;
   }
 
-  static HttpClientConnectionManager configureConnectionManager(Map<String, String> properties) {
+  static HttpClientConnectionManager configureConnectionManager(
+      Map<String, String> properties, TlsSocketStrategy tlsSocketStrategy) {
     PoolingHttpClientConnectionManagerBuilder connectionManagerBuilder =
         PoolingHttpClientConnectionManagerBuilder.create();
     ConnectionConfig connectionConfig = configureConnectionConfig(properties);
     if (connectionConfig != null) {
       connectionManagerBuilder.setDefaultConnectionConfig(connectionConfig);
     }
+    if (tlsSocketStrategy != null) {
+      connectionManagerBuilder.setTlsSocketStrategy(tlsSocketStrategy);
```

Comment:
typically stuff like this is all configured via the `properties` map. Otherwise there isn't a way to configure this e.g. when using the REST catalog with Spark

Reply:
@nastra 
The idea was similar to how [httpProxy](https://github.com/apache/iceberg/blob/f9b9621e77ef44d4e5fd72bf6014e3037514ba98/core/src/main/java/org/apache/iceberg/rest/HTTPClient.java#L455-L461) is supported. HttpProxy can be configured via system properties (note: it is not configurable via properties map) or via explicitly specifying host and port where applicable (might not be possible in case of spark). Same in case of `ProxyCredentialProvider` too

Similarly, here, I'm adding an additional way to configure tlsSocketStrategy where more granular control is required like custom keystore to construct sslContext and so on.

Reply:
ok this is probably fine to just be configured through the builder. I'm ok with this change but let's see what other reviewers think of this